units/host_allocator.go

```diff
@@ -27,6 +27,7 @@ const (
 	hostAllocatorJobName         = "host-allocator"
 	hostAllocatorAttributePrefix = "evergreen.host_allocator"
 	maxHostAllocatorJobTime      = 10 * time.Minute
+	maxIntentHosts               = 5000
```

Comment:
Could you add a comment here about the impact of this? e.g. once hit, this will cause host allocator jobs to no-op successfully. Is there anything in the future we should think about when changing the number? (I only mention to comment it now, while you have the most context about it)